src/sb-util.ts

```diff
@@ -221,43 +226,78 @@ export class Sprite extends SpriteCollection {
     /**
      * @return {VariableCollection}. The variables available to a Sprite include the
      *      variables attached to the sprite as well as variables in the
-     *      global scope, which are attached to the stage
+     *      global scope, which are attached to the stage. Also included in variables
+     *      are broadcasts (which are always global) and lists, which can be global
+     *      or local.
      */
     public variables(): VariableCollection {
         let taggedVariables: Iterable<VariableProperties>;
 
+        function setMetaSprite(
+            deserialized: Iterable<VariableProperties>,
+            sprite: SpriteProperties,
+        ): Iterable<VariableProperties> {
+            return setMetaIterable(
+                deserialized,
+                (v): VariableProperties => {
+                    getVariableMeta(v).sprite = sprite;
+                    return v;
+                },
+            );
+        }
+
         const stage = getSpriteMeta(this.props()).project.stage();
```

Comment:
Logically - I think this would make more sense to me if it did all three (broadcast, list, variables) on itself first.

Rather than operating on stage for these first three, operate on `this` instead, then after you build `taggedVariables` out of those 3 iterables, you can do:

```js
if (!this.isStage()) {
  taggedVariables = chain(taggedVaraibles, getSpriteMeta(this.props()).project.stage().variables().propsIterable());
}

return new Variables(taggedVariables)